src/PowerShellEditorServices.Protocol/Server/DebugAdapter.cs

```diff
@@ -186,15 +187,45 @@ protected override void Shutdown()
             await requestContext.SendResult(null);
         }
 
-        protected Task HandleAttachRequest(
+        protected async Task HandleAttachRequest(
             AttachRequestArguments attachParams,
             RequestContext<object> requestContext)
         {
-            // TODO: Implement this once we support attaching to processes
-            throw new NotImplementedException();
+            this.isAttach = true;
+
+            if (attachParams.ProcessId > 0)
+            {
+                // TODO: Version check!
+
+                await this.editorSession.PowerShellContext.ExecuteScriptString(
+                    $"Enter-PSHostProcess -Id {attachParams.ProcessId}");
+
+                if (attachParams.RunspaceId > 0)
+                {
+                    // Execute the Debug-Runspace command but don't await it because it
+                    // will block the debug adapter initialization process.
+                    Task nonAwaitedTask =
```

Comment:
Yeah, the C# compiler gives a warning if you call an async method without doing something with the task.  I've been using this trick to silence the message in cases where I don't actually want to await the task.